### PR TITLE
chore: sync v3 with main branch

### PR DESCRIPTION
**Pull request by tswast**
"do not merge" is just a reminder. As with the last one, I'd like to use a merge commit.

I had to manually resolve a few conflicts, but not nearly as many as last time.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -366,8 +366,9 @@
     "grpc": ("https://grpc.github.io/grpc/python/", None),
     "proto-plus": ("https://proto-plus-python.readthedocs.io/en/latest/", None),
     "protobuf": ("https://googleapis.dev/python/protobuf/latest/", None),
-    "pandas": ("http://pandas.pydata.org/pandas-docs/stable/", None),
+    "dateutil": ("https://dateutil.readthedocs.io/en/latest/", None),
     "geopandas": ("https://geopandas.org/", None),
+    "pandas": ("https://pandas.pydata.org/pandas-docs/dev", None),
```
Comment (tswast) [inline docs/conf.py:371]: fixed via https://github.com/googleapis/python-bigquery/pull/1048

**Comment by plamut** [comment]
@googlebot I fixed it.

**Review by plamut — APPROVED** [review]
I think it's good.

What were the conflicts this time, were there any doubts when resolving them?

**Comment by tswast** [comment]
> What were the conflicts this time, were there any doubts when resolving them?

* Some dependencies in setup.py (pyarrow expansion), no doubts there.
* Some imports, due to adding dateutil, no doubts there.

**Pull request merged**